pkg/polaris-reporting/ctl_polarisreporting.go

```diff
@@ -76,52 +76,70 @@ func (ctl *ArgsFromCobraFlags) GetArgs() map[string]string {
 
 // AddFlagsToCommand adsf
 func (ctl *ArgsFromCobraFlags) AddFlagsToCommand(cmd *cobra.Command, master bool) {
-	// [DEV NOTE:] please organize flags in order of importance
-	cmd.Flags().StringVar(&ctl.flagTree.Version, "version", ctl.flagTree.Version, "Version of Polaris you want to install [Example: \"2019.11\"]\n")
+	// [DEV NOTE:] please organize flags in order of importance and group related flags
+	cmd.Flags().StringVar(&ctl.flagTree.Version, "version", ctl.flagTree.Version, "Version of Polaris-Reporting you want to install [Example: \"0.0.12\"]\n") // TODO: Put a real version here
 
-	// domain-name specific flags
-	cmd.Flags().StringVar(&ctl.flagTree.GCPServiceAccount, "gcp-service-account-path", ctl.flagTree.GCPServiceAccount, "Absolute path to given Google Cloud Platform service account for pulling images")
-	cobra.MarkFlagRequired(cmd.Flags(), "gcp-service-account-path")
+	// domain specific flags
+	cmd.Flags().StringVar(&ctl.flagTree.FQDN, "fqdn", ctl.flagTree.FQDN, "Fully qualified domain name [Example: \"example.polaris.synopsys.com\"]\n")
 	// cmd.Flags().StringVar(&ctl.flagTree.IngressClass, "ingress-class", "", "Name of ingress class")
-	cmd.Flags().StringVar(&ctl.flagTree.FQDN, "fqdn", "", "Fully qualified domain name [Example: \"example.polaris.synopsys.com\"]\n")
-	cobra.MarkFlagRequired(cmd.Flags(), "fqdn")
+
+	if master {
+		cobra.MarkFlagRequired(cmd.Flags(), "fqdn")
+	}
+
+	// license related flags
+	if master {
```

Comment:
why so many if master checks?

Reply:
I did it that way to keep the flags organized and for code readability/maintainability